strawberryfields/api/connection.py

```diff
@@ -0,0 +1,284 @@
+        """The API authentication token.
+
+        Returns:
+            str: the authentication token
+        """
+        return self._token
+
+    @property
+    def host(self) -> str:
+        """The host for the remote platform.
+
+        Returns:
+            str: the hostname
+        """
+        return self._host
+
+    @property
+    def port(self) -> int:
+        """The port to connect to on the remote host.
+
+        Returns:
+            int: the port number
+        """
+        return self._port
+
+    @property
+    def use_ssl(self) -> bool:
+        """Whether to use SSL for the connection.
+
+        Returns:
+            bool: ``True`` if SSL should be used, and ``False`` otherwise
+        """
+        return self._use_ssl
+
+    def create_job(self, target: str, program: Program, shots: int) -> Job:
+        """Creates a job with the given circuit.
+
+        Args:
+            target (str): the target device
+            program (strawberryfields.Program): the quantum circuit
+            shots (int): the number of shots
+
+        Returns:
+            strawberryfields.engine.Job: the created job
+        """
+        # Serialize a blackbird circuit for network transmission
+        bb = to_blackbird(program)
+        # pylint: disable=protected-access
+        bb._target["name"] = target
+        # pylint: disable=protected-access
+        bb._target["options"] = {"shots": shots}
+        circuit = bb.serialize()
+
+        path = "/jobs"
+        response = requests.post(
+            self._url(path),
+            headers=self._headers,
+            data=json.dumps({"circuit": circuit}),
+        )
+        if response.status_code == 201:
+            if self._verbose:
+                log.info("The job was successfully submitted.")
+            return Job(
+                id_=response.json()["id"],
+                status=JobStatus(response.json()["status"]),
+                connection=self,
+            )
+        raise RequestFailedError(
+            "Failed to create job: {}".format(self._format_error_message(response))
+        )
+
+    def get_all_jobs(self, after: datetime = datetime(1970, 1, 1)) -> List[Job]:
+        """Gets a list of jobs created by the user, optionally filtered by datetime.
+
+        Args:
+            after (datetime.datetime): if provided, only jobs more recently created
+                then ``after`` are returned
+
+        Returns:
+            List[strawberryfields.engine.Job]: the jobs
+        """
+        raise NotImplementedError("This feature is not yet implemented")
+
+    def get_job(self, job_id: str) -> Job:
+        """Gets a job.
+
+        Args:
+            job_id (str): the job ID
+
+        Returns:
+            strawberryfields.engine.Job: the job
+        """
+        path = "/jobs/{}".format(job_id)
+        response = requests.get(self._url(path), headers=self._headers)
+        if response.status_code == 200:
+            return Job(
```

Comment:
Might be worthwhile to make this `Job`(...)` block a separate function, since it is used elsewhere in the same way

Reply:
Do you mean the `Job(...)` "constructor"? Since it is already is (sort of) a function in itself, I'm not sure I see the benefit of factoring it out further, unless I'm missing something :thinking: